### PR TITLE
Add separate methods for creating streams

### PR DESCRIPTION
**Pull request by shadowhand**
Being able to create streams from 3 distinct sources aligns with
existing middleware that may need to:

- Create streams from strings
- Create streams from files
- Create streams from resources

In the case of strings and files, the final result will be a resource
that will be attached to the stream.

Refs #2
Refs #4
Refs #14
Refs #16

### CODE REVIEW

**Comment by shadowhand** [comment]
Are we satisfied with this approach @http-interop/http-middleware-contributors ?

File: src/StreamFactoryInterface.php [hunk 1]
```diff
@@ -7,17 +7,37 @@
...
      *
      * @return StreamInterface
      */
-    public function createStream($resource);
+    public function createStream($content = '');
+
+    /**
+     * Create a stream from an existing file.
+     *
+     * The file must be opened using the given mode, which may be any mode
```
Comment (Nyholm) [inline src/StreamFactoryInterface.php]: MUST

File: src/StreamFactoryInterface.php [hunk 2]
```diff
@@ -7,17 +7,37 @@
...
+     * @param string $mode
+     *
+     * @return StreamInterface
+     */
+    public function createStreamFromFile($file, $mode = 'r');
+
+    /**
+     * Create a new stream from an existing resource.
+     *
+     * The stream must be readable and may be writable.
```
Comment (Nyholm) [inline src/StreamFactoryInterface.php]: MUST and MAY

File: src/StreamFactoryInterface.php [hunk 3]
```diff
@@ -7,17 +7,37 @@
...
     /**
-     * Create a new stream from a resource or a string.
+     * Create a new stream from a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * The stream should be created with a temporary resource.
```
Comment (Nyholm) [inline src/StreamFactoryInterface.php]: SHOULD

**Review by Nyholm — APPROVED** [review]
I think this sums up everything. Thank you. 

I had some comments about using uppercase.

**Comment by shadowhand** [comment]
Merged. If anyone has concerns, please comment here and I will address them in a separate PR.

**Comment by mindplay-dk** [comment]
I'm still not thrilled about having one method called `createStream` when all other methods have names that qualify how the stream is made.

Meh, close enough, I guess...

**Pull request merged**